src/commands/signzone.rs

```diff
@@ -0,0 +1,1782 @@
+        let zonemd_rrs: SortedRecords<StoredName, StoredRecordData> =
+            SortedRecords::from(zonemd_rrs);
+        // No need to check for keys, as SortedRecords::sign just doesn't do anything without keys.
+        let mut zonemd_rrsig = zonemd_rrs
+            .sign(
+                apex,
+                self.expiration,
+                self.inception,
+                keys.as_slice(),
+                false,
+            )
+            .unwrap();
+
+        // Replace original ZONEMD RRSIG with newly generated one
+        if let Some(rrsig) = zonemd_rrsig.pop() {
+            if let ZoneRecordData::Rrsig(rrsig) = rrsig.data() {
+                records.replace_rrsig_for_apex_zonemd(rrsig.clone(), apex);
+            }
+        }
+    }
+}
+
+fn next_owner_hash_to_name(
+    next_owner_hash_hex: &str,
+    apex: &FamilyName<Name<Bytes>>,
+) -> Result<Name<Bytes>, ()> {
+    let mut builder = NameBuilder::new_bytes();
+    builder
+        .append_chars(next_owner_hash_hex.chars())
+        .map_err(|_| ())?;
+    let next_owner_name = builder.append_origin(apex.owner()).map_err(|_| ())?;
+    Ok(next_owner_name)
+}
+
+//------------ SigningMode ---------------------------------------------------
+
+#[derive(Copy, Clone, Debug, Default, PartialEq, Eq)]
+enum SigningMode {
+    /// Both hash (NSEC/NSEC3) and sign zone records.
+    #[default]
+    HashAndSign,
+
+    /// Only hash (NSEC/NSEC3) zone records, don't sign them.
+    HashOnly,
+    // /// Only sign zone records, assume they are already hashed.
+    // SignOnly,
+    /// Neither hash or sign zone records (e.g. when just using ZONEMD).
+    None,
+}
+
+//------------ ZonemdTuple ---------------------------------------------------
+
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+struct ZonemdTuple(ZonemdScheme, ZonemdAlgorithm);
+
+//------------ FileOrStdout --------------------------------------------------
+
+enum FileOrStdout<T: io::Write, U: fmt::Write> {
+    File(T),
+    Stdout(Stream<U>),
+}
+
+impl<T: io::Write, U: fmt::Write> fmt::Write for FileOrStdout<T, U> {
+    fn write_str(&mut self, s: &str) -> std::fmt::Result {
+        match self {
+            FileOrStdout::File(f) => f.write_all(s.as_bytes()).map_err(|_| fmt::Error),
+            FileOrStdout::Stdout(o) => {
+                o.write_str(s);
+                Ok(())
+            }
+        }
+    }
+
+    fn write_fmt(&mut self, args: fmt::Arguments<'_>) -> fmt::Result {
+        match self {
+            FileOrStdout::File(f) => f.write_fmt(args).map_err(|_| fmt::Error),
+            FileOrStdout::Stdout(o) => {
+                o.write_fmt(args);
+                Ok(())
+            }
+        }
+    }
+}
+
+//------------ Commented -----------------------------------------------------
+
+/// Support for RTYPE specific zonefile comment generation.
+///
+/// Intended to be used to enable behaviour to be matched to that of the LDNS
+/// `ldns_rr2buffer_str_fmt()` function.
+trait Commented<T> {
```

Comment:
Hmm, I just noticed that we have comment support in `ZonefileFmt` in `domain`. That should be used instead of this I guess.